src/servers/src/context.rs

```diff
@@ -12,26 +12,25 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::collections::HashMap;
 use std::sync::Arc;
 
-use serde::{Deserialize, Serialize};
 use snafu::OptionExt;
 
 use crate::error::{BuildingContextSnafu, Result};
 
 type CtxFnRef = Arc<dyn Fn(&Context) -> bool + Send + Sync>;
 
-#[derive(Serialize, Deserialize)]
 pub struct Context {
-    pub exec_info: ExecInfo,
     pub client_info: ClientInfo,
     pub user_info: UserInfo,
     pub quota: Quota,
-    #[serde(skip)]
     pub predicates: Vec<CtxFnRef>,
 }
 
+pub struct UserInfo {
```

Comment:
what's the difference of this `UserInfo` and the one in `auth` file?

can that one be reused here?